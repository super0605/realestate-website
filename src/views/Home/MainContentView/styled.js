import styled from "styled-components";

export const Container = styled.div`
  padding: ${({ theme }) => theme.spacings.spacing_xl};
  padding-top: ${({ theme }) => theme.spacings.spacing_xl};
  height: calc(100vh - 150px);
`;

export const NoDataView = styled.div`
  font-size: 30px;
  font-weight: bold;
  font-family: "Roboto";
  color: #d9d9d9;
  width: 100%;
  min-height: 200px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
