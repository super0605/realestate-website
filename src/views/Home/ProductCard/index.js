import React from "react";
import PropTypes from "prop-types";
import NumberFormat from "react-number-format";
import {
  Container,
  ProductImg,
  Descriptoin,
  ValueGroup,
  ValueElm,
  DividerH,
  Overlay,
  OverlayContent,
  Price,
  PricePerSqm,
} from "./styled";

const ProductCard = ({
  title,
  price,
  pricePerSqm,
  sqm,
  numberOfBedrooms,
  numberOfBathrooms,
  imgUrl,
}) => {
  return (
    <Container>
      <ProductImg>
        {imgUrl && <img alt="example" src={imgUrl} />}
        <Overlay>
          <OverlayContent>
            {price && (
              <Price>
                <NumberFormat
                  displayType={"text"}
                  value={price}
                  thousandSeparator={true}
                  decimalScale={0}
                  suffix="€"
                />
              </Price>
            )}
            {pricePerSqm && (
              <PricePerSqm>
                <NumberFormat
                  displayType={"text"}
                  value={pricePerSqm}
                  thousandSeparator={true}
                  decimalScale={3}
                  suffix="€/m²"
                />
              </PricePerSqm>
            )}
          </OverlayContent>
        </Overlay>
      </ProductImg>
      {title && <Descriptoin>{title}</Descriptoin>}
      <ValueGroup>
        {sqm && (
          <ValueElm>
            <i className="fa fa-check-circle" /> {sqm} m²
          </ValueElm>
        )}
        <DividerH />
        {numberOfBedrooms ? (
          <ValueElm>
            <i className="fa fa-bed" /> {numberOfBedrooms} habs.
          </ValueElm>
        ) : (
          <ValueElm>N/A</ValueElm>
        )}
        <DividerH />
        {numberOfBathrooms ? (
          <ValueElm>
            <i className="fa fa-bath" /> {numberOfBathrooms} banos
          </ValueElm>
        ) : (
          <ValueElm>N/A</ValueElm>
        )}
      </ValueGroup>
    </Container>
  );
};

ProductCard.propTypes = {
  title: PropTypes.string,
  price: PropTypes.number,
  pricePerSqm: PropTypes.number,
  sqm: PropTypes.number,
  numberOfBedrooms: PropTypes.number,
  numberOfBathrooms: PropTypes.number,
  imgUrl: PropTypes.string,
};

export default ProductCard;
