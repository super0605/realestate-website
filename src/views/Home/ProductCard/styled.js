import styled from "styled-components";
import Divider from "../../../components/Divider";

export const Container = styled.div`
  background-color: #ffffff;
  min-height: 143px;
  margin-bottom: 24px;
  cursor: pointer;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.06), 0 1px 0 0 rgba(0, 0, 0, 0.07),
    0 2px 1px -1px rgba(0, 0, 0, 0.06);

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 10px, rgba(0, 0, 0, 0.15) 10px 10px 10px;
  }
`;

export const ProductImg = styled.div`
  width: 100%;
  height: 200px;
  background-color: #ffffff;
  min-height: 150px;
  position: relative;
  & img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

export const Descriptoin = styled.div`
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 400;
  padding: 10px;
`;

export const ValueGroup = styled.div`
  border-top: 1px solid #d9d9d9;
  padding: 14px 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const ValueElm = styled.div`
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 600;
`;

export const DividerH = styled(Divider).attrs(({ theme }) => ({
  height: "20px",
  width: "2px",
  color: "#d9d9d9",
}))``;

export const Overlay = styled.div`
  position: absolute;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.8);
  width: 100%;
  padding: 10px;
`;

export const OverlayContent = styled.div``;

export const Price = styled.div`
  color: #000000;
  font-size: 18px;
  font-family: "Roboto";
  font-weight: 600;
`;
export const PricePerSqm = styled.div`
  color: #000000;
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 400;
`;
