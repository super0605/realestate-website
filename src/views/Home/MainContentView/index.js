import React from "react";
import { debounce } from "lodash";
import { Row, Col } from "antd";
import PerfectScrollbar from "react-perfect-scrollbar";
import ProductCard from "../ProductCard";
import { Container } from "./styled";

const MainContentView = ({ apartments, onLoadMore }) => {
  const handlePagination = (scrollContainer) => {
    scrollContainer.scrollTop = scrollContainer.scrollTop - 30;
    if (apartments.length > 0) {
      onLoadMore();
    }
  };

  return (
    <PerfectScrollbar
      options={{ suppressScrollX: true, wheelPropagation: false, cancelable: false }}
      onYReachEnd={debounce(handlePagination, 100)}
    >
      <Container>
        <Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}>
          {apartments &&
            apartments.map((property, index) => (
              <Col key={index} className="gutter-row" xs={24} sm={12} md={12} lg={8} xl={6}>
                <ProductCard
                  title={property.title}
                  price={property.price}
                  pricePerSqm={property.pricePerSqm}
                  sqm={property.sqm}
                  numberOfBedrooms={property.numberOfBedrooms}
                  numberOfBathrooms={property.numberOfBathrooms}
                  imgUrl={property.picture}
                />
              </Col>
            ))}
        </Row>
      </Container>
    </PerfectScrollbar>
  );
};

export default MainContentView;
