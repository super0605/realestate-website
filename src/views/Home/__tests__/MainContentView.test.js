import React from "react";
import { mount, shallow } from "enzyme";
import MainContentView from "../MainContentView";
import ProductCard from "../ProductCard";

describe("MainContentView", () => {
  let props;
  let mountedMainContentView;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedMainContentView) {
      mountedMainContentView = bShallow
        ? shallow(<MainContentView {...props} />)
        : mount(<MainContentView {...props} />);
    }
    return mountedMainContentView;
  };

  beforeEach(() => {
    props = {
      apartments: [
        {
          id: "QXBhcnRtZW50LTEwMDA=",
          numberOfBathrooms: 2,
          numberOfBedrooms: 3,
          picture: "house.jpg",
          price: 30000,
          pricePerSqm: 4000,
          sqm: 120,
          title: "Lillian West",
          __typename: "Apartment",
        },
      ],
    };

    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render product card", () => {
    expect(wrapper.find(ProductCard).length).toBe(1);
  });
});
