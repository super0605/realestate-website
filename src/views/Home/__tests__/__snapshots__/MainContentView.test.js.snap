// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContentView should render without problems 1`] = `
<ScrollBar
  className=""
  component="div"
  containerRef={[Function]}
  onSync={[Function]}
  onYReachEnd={[Function]}
  options={
    Object {
      "suppressScrollX": true,
    }
  }
>
  <styled.div>
    <Row
      gutter={
        Object {
          "lg": 32,
          "md": 24,
          "sm": 16,
          "xs": 8,
        }
      }
    >
      <Col
        className="gutter-row"
        key="0"
        lg={8}
        md={12}
        sm={12}
        xl={6}
        xs={24}
      >
        <ProductCard
          imgUrl="house.jpg"
          numberOfBathrooms={2}
          numberOfBedrooms={3}
          price={30000}
          pricePerSqm={4000}
          sqm={120}
          title="Lillian West"
        />
      </Col>
    </Row>
  </styled.div>
</ScrollBar>
`;
